blockprod/src/detail/block_maker.rs

```diff
@@ -207,8 +209,40 @@ impl BlockMaker {
         let accumulator = self.collect_transactions().await?;
         let transactions = accumulator.transactions();
 
+        // TODO: instead of the following static value, look at
+        // self.chain_config for the current block reward, then send
+        // it to self.reward_destination
+        let block_reward = BlockReward::new(vec![]);
+
+        let block_body = BlockBody::new(block_reward, transactions.clone());
+
+        let tx_merkle_root =
+            calculate_tx_merkle_root(&block_body).map_err(BlockCreationError::MerkleTreeError)?;
+        let witness_merkle_root = calculate_witness_merkle_root(&block_body)
+            .map_err(BlockCreationError::MerkleTreeError)?;
+
+        let stop_flag = Arc::new(false.into());
+
         loop {
-            let block = self.generate_block(transactions.clone()).await?;
+            let timestamp = BlockTimestamp::from_duration_since_epoch(self.time_getter.get_time());
+            let consensus_data = self.pull_consensus_data(self.current_tip_id, timestamp).await?;
+
+            let block_header = BlockHeader::new(
+                self.current_tip_id,
+                tx_merkle_root,
+                witness_merkle_root,
+                timestamp,
+                consensus_data,
+            );
+
```

Comment:
The above three assignments don't change per iteration, so I've moved them in my PR.